Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -99,12 +127,44 @@ public float Distance
         private int rows = 3;
 
         /// <summary>
-        /// Number of rows per column, column number is automatically determined
+        /// Number of rows per column.false Can only be assigned when layout type is
+        /// RowsThenColumns
         /// </summary>
         public int Rows
         {
             get { return rows; }
-            set { rows = value; }
+            set 
+            {
+                if (Layout == LayoutOrder.ColumnThenRow) 
+                {
+                    Debug.LogError("Cannot assign Rows when layout type is " + Layout);
```

Comment:
```suggestion
                    Debug.LogError("Cannot assign Rows when layout type is ColumnThenRow");
```

Reply:
Nit: In the particular case where we are trying to log a specific error message for a single case (i.e. ColumnThenRow), it's more helpful to have a non-concatenated string. (i.e. Layout in this case will always only be a single type so that imagine...

that someone is searching for "Cannot assign Rows when layout type is ColumnThenRow" they can find the exact string in the code.

Reply:
Actually wait, reading this closer, I think it might be more helpful to give an error saying what IS expected instead of what ISN'T expected (i.e. this should only be used with RowsThenColumns, so that it tells people how to use it correctly, rather than telling them they are using it wrong without a direct guide on how to use it right).

i.e. I think a general principle is if we're going to give an error message, we should tell people how to get things right, instead of how they got things wrong.

Reply:
Fixed, great suggestion